src/crypto/ecc_curve.rs

```diff
@@ -191,6 +203,17 @@ mod tests {
         assert_eq!(ecc_curve_from_oid(vec![1, 2, 3].as_slice()), None);
     }
 
+    #[test]
+    fn test_unknown_ecc_curve() {
+        // brainpoolP160r1 (an OID we don't handle explicitly in ECCCurve)
+        const OID: &[u8] = &[
+            0x06, 0x09, 0x2B, 0x24, 0x03, 0x03, 0x02, 0x08, 0x01, 0x01, 0x01,
+        ];
+
+        let bp_p160r1 = ecc_curve_from_oid(OID).unwrap();
+        assert_eq!(bp_p160r1.oid_str(), "1.3.36.3.3.2.8.1.1.1");
```

Comment:
This test seems to be misinterpreting a DER header as part of the OID.

I'm not sure how `ecc_curve_from_oid` is defined, but it seems like you are using the parser for an OID body, but giving it a DER-encoded OID instead of just the body, i.e. with a header containing the leading `0x06` = OID tag, and `0x09` length octets.

If you want to parse a DER encoded OID, you need to use `ObjectIdentifier::from_der`. Otherwise, strip the leading first two bytes since they contain a DER header.

Reply:
Oh. I see, thank you for the explanation!

Reply:
I've dropped the DER header from the test-OID, and of course that fixes the issue.